test_midas.py

```diff
@@ -0,0 +1,170 @@
+    assert event is not None
+    assert event.location == new_event.location
+    assert event.date == new_event.date
+
+
+def test_organization_members(db_session):
+    organization = midas.get_organization(db_session, organizations[0])
+    assert len(organization.members) == 4
+    assert organization.members[0].id == terrorists[0]
+    assert organization.members[1].id == terrorists[1]
+    assert organization.members[2].id == terrorists[4]
+    assert organization.members[3].id == terrorists[5]
+
+
+def test_terrorist_organization(db_session):
+    terrorist = midas.get_terrorist(db_session, terrorists[0])
+    assert terrorist.organization.id == organizations[0]
+
+
+def test_event_participants(db_session):
+    event = midas.get_event(db_session, events[0])
+    assert len(event.participants) == 3
+    assert event.participants[0].id == terrorists[0]
+    assert event.participants[1].id == terrorists[1]
+    assert event.participants[2].id == terrorists[2]
+
+
+def test_get_members_not_kalab(db_session):
+    not_kalab = midas.get_members_not_kalab(db_session)
+    actual_not_kalab = {1, 2, 3, 6}
+    assert len(not_kalab) == len(actual_not_kalab)
+    assert len(set.intersection(set(not_kalab), actual_not_kalab)) == len(actual_not_kalab)
+
+
+def test_last_event_participated_in(db_session):
+    terrorist_last_event_date = midas.get_last_event_participated_in(db_session)
+    assert len(terrorist_last_event_date) == len(terrorists)
+    assert terrorist_last_event_date[terrorists[4]] is None
+    event_date = midas.get_event(db_session, events[1]).date
+    assert terrorist_last_event_date[terrorists[0]] == event_date
+
+
+def test_get_organizations_members_count(db_session):
+    organization_members_count = midas.get_organizations_members_count(db_session)
+    assert len(organization_members_count) == len(organizations)
+    assert organization_members_count[organizations[0]] == 4
+    assert organization_members_count[organizations[1]] == 2
+
+
+def test_get_organizations_count_per_event(db_session):
+    organization_count_per_event = midas.get_organizations_count_per_event(db_session)
+    assert len(organization_count_per_event) == len(events)
+    assert organization_count_per_event[events[0]] == 2
+    assert organization_count_per_event[events[1]] == 2
+    assert organization_count_per_event[events[2]] == 1
+    assert organization_count_per_event[events[3]] == 1
+
+
+def test_people_you_may_know(db_session):
+    people_you_may_know = midas.get_people_you_may_know(db_session)
+    assert len(people_you_may_know) == len(terrorists)
+    assert len(people_you_may_know[terrorists[0]]) == 3
+    assert len(people_you_may_know[terrorists[1]]) == 3
+    assert len(people_you_may_know[terrorists[2]]) == 2
+    assert len(people_you_may_know[terrorists[3]]) == 1
+    assert len(people_you_may_know[terrorists[4]]) == 0
+    assert len(people_you_may_know[terrorists[5]]) == 1
+
+
+def test_refine_functionality(db_session):
```

Comment:
Split this test into several test cases.
Try and test all the features requested of the `refine` method.

Reply:
Done